src/pnpq/errors.py

```diff
@@ -63,3 +66,83 @@ class InvalidStateException(Exception):
     re-open them.
 
     """
+
+
+class ThorlabsOswError(Exception):
+    """Exception raised for errors reported by Thorlabs OSWxx-yyyyE devices.
+
+    The instance provides three attributes: ``code`` (int or None),
+    ``description`` (str), and ``raw_reply`` (str).
+    """
+
+    def __init__(
+        self,
+        code: Optional[int],
```

Comment:
There should be an open-ended enum that names the error codes. This shouldn't be an int.